### PR TITLE
more coroutines

### PR DESCRIPTION
**Pull request by havardpe**
@vekterli please review

Do not merge just yet. I am seeing local valgrind errors complaining about leaked tls storage or something.

### CODE REVIEW

File: vespalib/src/vespa/vespalib/coro/lazy.h [hunk 1]
```diff
@@ -5,10 +5,23 @@
...
 
 namespace vespalib::coro {
 
+/**
+ * coroutine return type
+ *
+ * The coroutine is lazy (will suspend in initial_suspend) and
+ * destroyed form the outside (will suspend in final_suspend). Waiting
+ * for a Lazy<T> using co_await will use symmetric transfer to suspend
+ * the waiting coroutine and resuming this one. The waiting coroutine
```
Comment (vekterli) [inline vespalib/src/vespa/vespalib/coro/lazy.h:19]:
```suggestion
 * the waiting coroutine and resume this one. The waiting coroutine
```

File: vespalib/src/vespa/vespalib/coro/lazy.h [hunk 2]
```diff
@@ -5,10 +5,23 @@
...
+ * is registered as a continuation and will be resumed again once the
+ * result is avalable (also using symmetric transfer). The result is
```
Comment (vekterli) [inline vespalib/src/vespa/vespalib/coro/lazy.h:21]:
```suggestion
 * result is available (also using symmetric transfer). The result is
```

**Review by vekterli — APPROVED** [review]
✨🦄👍

**Comment by havardpe** [comment]
@vekterli PTAL

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**